### PR TITLE
[Xwt.XamMac] Add application dock menu support to AppDelegate

### PR DESCRIPTION
**Pull request by iainx**
Adds an event to the AppDelegate to return custom dock menu items

### CODE REVIEW

File: Xwt.XamMac/Xwt.Mac/MacEngine.cs [hunk 1]
```diff
@@ -376,4 +390,9 @@ public OpenUrlEventArgs (string url)
 			Url = url;
 		}
 	}
+
+	public class ShowDockMenuArgs : EventArgs
+	{
+		public NSMenu dockMenu;
```
Comment (slluis) [inline Xwt.XamMac/Xwt.Mac/MacEngine.cs]: This should be a property, and first letter should be upper case

Comment (iainx) [inline Xwt.XamMac/Xwt.Mac/MacEngine.cs]: Yeah, not sure what I was thinking here. Fixed it in the new push

**Comment by sevoku** [comment]
@iainx whats the status here? Could it be still useful, even if not used in MD?

**Comment by Therzok** [comment]
Ping on this, would be good to merge so we can take a look at:
https://github.com/mono/monodevelop/pull/1065

**Comment by sevoku** [comment]
LGTM, lets merge. This could be also useful for a XWT wide application menu, Gtk has also support for application menus. We'd need an api for a main app window...

**Pull request merged**